libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationHelper.java

```diff
@@ -299,7 +300,7 @@ static NavigationIndices increaseIndex(RouteProgress routeProgress,
                                                                              List<StepIntersection> intersections) {
     List<Pair<StepIntersection, Double>> distancesToIntersections = new ArrayList<>();
     List<StepIntersection> stepIntersections = new ArrayList<>(intersections);
-    if (stepPoints.isEmpty()) {
+    if (stepPoints.size() < TWO_POINTS) {
```

Comment:
What about consolidating these two guard clauses 👇 (that return the same) extracting the conditional into a explaining variable?
Also what about creating `stepIntersections` after the guard clauses?

> **Source code structure**
> Variables and methods should be defined close to where they are used.
> Local variables should be declared just above their first usage and should
have a small vertical scope.

_Clean Code_ (Robert C. Martin)

That way we avoid allocating it if any of the guard clauses is executed.